go/database/mpt/proof_test.go

```diff
@@ -0,0 +1,462 @@
+package mpt
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/Fantom-foundation/Carmen/go/common"
+	"github.com/Fantom-foundation/Carmen/go/database/mpt/shared"
+	"go.uber.org/mock/gomock"
+	"math/rand"
+	"slices"
+	"testing"
+)
+
+func TestCreateAccountProof_CanGenerateProof(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	nodes := []Node{&BranchNode{}, &ExtensionNode{}, &AccountNode{address: common.Address{0xA}}}
+
+	nodeSource := mockNodeSource(ctrl, nodes)
+	hasher := nodeSource.getConfig().Hashing.createHasher()
+	root := NewNodeReference(EmptyId())
+	source := NewProofSource(&root, nodeSource)
+
+	proof, err := source.CreateAccountProof(common.Address{0xA})
+	if err != nil {
+		t.Fatalf("failed to create account proof: %v", err)
+	}
+
+	for i, element := range proof {
+		rlp, err := hasher.encode(nodes[i], nodeSource, []byte{})
+		if err != nil {
+			t.Fatalf("failed to encode node: %v", err)
+		}
+		if got, want := element.Rlp(), rlp; !bytes.Equal(got, want) {
+			t.Errorf("unexpected RLP: got %x, want %x", got, want)
+		}
+		if got, want := element.Hash(), common.Keccak256(rlp); got != want {
+			t.Errorf("unexpected hash: got %x, want %x", got, want)
+		}
```

Comment:
The Proof types should eventually have a "IsValid" function which is checking this. Implementing this first, would simplify those tests.